test-cypress/integration/sparql-editor/sparql-editor.spec.js

```diff
@@ -52,4 +52,38 @@ describe('Sparql editor', () => {
         // And yasr tabs should not be visible
         YasrSteps.getResultHeader().should('not.be.visible');
     });
+
+    it('Should paste and add prefixes', () => {
+        // Given I have opened the workbench
+        // When I visit the sparql editor page
+        SparqlEditorSteps.visitSparqlEditorPage();
+        YasqeSteps.clearEditor();
+        // I paste a query into the editor
+        cy.pasteIntoCodeMirror('.CodeMirror', 'select distinct ?x ?Person  where {\n' +
+            '?x a afn:Person .\n' +
+            '?x afn:preferredLabel ?Person .\n' +
+            '?doc afn:containsMention / pub-old:hasInstance ?x .\n' +
+            '}');
+        // Then I expect the prefixes to be added automatically
+        cy.get('.CodeMirror').then((codeMirrorElement) => {
+            const codeMirror = codeMirrorElement[0].CodeMirror;
+            // Wait, so that Yasqe can have time to replace the existing query with the new one
+            cy.waitUntil(() => {
+                return codeMirror.getValue() === 'PREFIX afn: <http://jena.apache.org/ARQ/function#>\n' +
+                    'select distinct ?x ?Person  where {\n' +
+                    '?x a afn:Person .\n' +
+                    '?x afn:preferredLabel ?Person .\n' +
+                    '?doc afn:containsMention / pub-old:hasInstance ?x .\n' +
+                    '}';
+            }).then(() => {
+                const value = codeMirror.getValue();
+                expect(value).to.equal('PREFIX afn: <http://jena.apache.org/ARQ/function#>\n' +
```

Comment:
So, you are checking the query twice now. Once in the wait function and then here. You can at least extract the query as a string template.